### PR TITLE
Add pixels to the ad click attribution mechanism

### PR DESCRIPTION
**Pull request by kzar**
This adds "pixel" requests to the mechanism we rely on to support ad
click attribution in DuckDuckGo products. These help DuckDuckGo
validate that the mechanism's logic is working correctly.

Further reading:
 - https://duckduckgo.com/duckduckgo-help-pages/privacy/web-tracking-protections/#duckduckgo-private-search-ads
 - https://improving.duckduckgo.com/

**Reviewer:** @sammacbeth , @jonathanKingston 

## Steps to test this PR:
1. Ensure the ad click attribution pixels fire correctly on Chrome using [the test page](https://www.search-company.site/) .
2. Ensure the pixels _don't_ fire on Firefox.
3. Ensure the integration tests and unit tests are passing.

## Automated tests:
- [x] Unit tests
- [x] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: integration-test/fire-button.spec.js [hunk 1]
```diff
@@ -134,7 +134,7 @@ test.describe('Fire Button', () => {
                 name: 'Last7days',
                 descriptionStats: {
                     clearHistory: true,
-                    cookies: 3, // gets the number of domains with cookies set
+                    cookies: 2, // gets the number of domains with cookies set
```
Comment (kzar) [inline integration-test/fire-button.spec.js:137]: @sammacbeth I'm not sure why this is necessary, seems to be unrelated the changes here?

File: integration-test/data/click-attribution-pixels.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
+// Expected pixel requests fired for each test case.
+// TODO: Move this into the shared testCases.json file.
+//       See https://github.com/duckduckgo/privacy-test-pages/blob/main/adClickFlow/shared/testCases.json
+exports.expectedPixels = [
```
Comment (kzar) [inline integration-test/data/click-attribution-pixels.js:4]: Note this file was reviewed already by @englehardt.

File: shared/js/background/devbuild.js [hunk 3]
```diff
@@ -37,7 +40,8 @@ export default function initDebugBuild () {
         setListContents,
         getListContents,
         companies: Companies,
-        ntts: createNewtabTrackerStatsDebugApi()
+        ntts: createNewtabTrackerStatsDebugApi(),
+        sendPageloadsWithAdAttributionPixelAndResetCount
```
Comment (sammacbeth) [inline shared/js/background/devbuild.js:44]: Was this just for debugging?

**Review by sammacbeth — APPROVED** [review]
Looks good, nice job! Just one thing I noticed that looks like something from debugging didn't get cleaned up.

File: shared/js/background/devbuild.js (hunk 3 above)
Comment (kzar) [reply shared/js/background/devbuild.js:44]: No, I exposed `sendPageloadsWithAdAttributionPixelAndResetCount` here so I could call it from the integration tests.
